src/index.ts

```diff
@@ -315,3 +441,32 @@ export function generateSalt(byteCount = 32): string {
   );
   return b64encoded;
 }
+
+/**
+ * Checks if the provided key is an `EncryptionKey`.
+ *
+ * @param key - The key to check.
+ * @returns Whether or not the key is an `EncryptionKey`.
+ */
+export function isEncryptionKey(
+  key: CryptoKey | EncryptionKey,
+): key is EncryptionKey {
+  return (
+    'key' in key &&
+    'derivationOptions' in key &&
+    'algorithm' in key.derivationOptions &&
+    'params' in key.derivationOptions
+  );
+}
+
+/**
+ * Checks if the provided key is an `ExportedEncryptionKey`.
+ *
+ * @param key - The key to check.
+ * @returns Whether or not the key is an `EncryptionKey`.
+ */
+export function isExportedEncryptionKey(
+  key: JsonWebKey | ExportedEncryptionKey,
+): key is ExportedEncryptionKey {
+  return 'key' in key && 'derivationOptions' in key;
+}
```

Comment:
Nit: Since we pass the result of `JSON.parse` into this function, would it be more accurate to say we don't know what this is?

``` suggestion
export function isExportedEncryptionKey(
  key: unknown,
): key is ExportedEncryptionKey {
  return (
    typeof key === 'object' &&
    key !== null &&
    'key' in key &&
    'derivationOptions' in key
  );
}
```